src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Snapshot/TargetedDependenciesSnapshot.cs

```diff
@@ -123,25 +170,50 @@ private bool FindUnresolvedDependenciesRecursive(IDependency dependency)
                     continue;
                 }
 
-                snapshotFilters.ForEach(
-                    filter => filter.BeforeRemove(ProjectPath, TargetFramework, dependency, worldBuilder, topLevelBuilder));
+                if (snapshotFilters != null)
+                {
+                    foreach (var filter in snapshotFilters)
+                    {
+                        dependency = filter.BeforeRemove(
+                            ProjectPath, TargetFramework, dependency, worldBuilder, topLevelBuilder, out bool filterAnyChanges);
+
+                        anyChanges |= filterAnyChanges;
```

Comment:
Nevermind. Didn't see the outer loop

---
In reply to: [111637938](https://github.com/dotnet/project-system/pull/1991#discussion_r111637938) [](ancestors = 111637938)